### DISCUSSION TITLE
Use StrimziPodSets concerns

### DISCUSSION THREAD

**Discussion by joseacl** [strimzi/strimzi-kafka-operator#7158] [Q&A] [answered]
We're using Strimzi Kafka Operator in our Kubernetes clusters to make the Kafka cluster deployment easier for development teams.
 
So far we're using the standard Statefulset set-up, disabling the feature gate of StrimziPodSet because we have the following concern:

The operator does not have HA, only deploy 1 pod. If we enable StrimziPodSet Feature Gate and for some reason the operator becomes unhealthy, we won't have anything that controls the Kafka cluster. Are we right? 
Is there a plan to keep the possibility of using Statefulset always instead of StrimziPodSet? 

Thanks in advance!

**Comment by scholzj** [comment, marked as answer]
If for some reason the operator does not run (or is for example in some broken state), then yes, the Pods will not be created. That is expected. Would being able to run multiple replicas of the cluster operator where one would be active and others will be waiting as a _warm backup_ mitigate your concerns?

We do not plan to keep supporting StatefulSets forever. They will be not used anymore in the future. Using the StrimziPodSets removes many issues and limitations caused by StatefulSets.

**Reply by joseacl** [reply]
Thanks for the quick response.
I see your point regarding not having support for statefulset in the future.
In general, I think having the possibility to set more than 1 replicas for the operator (using leader election pattern) would be useful.

**Reply by scholzj** [reply]
I opened #7174 to cover this. Please have a look if it corresponds to your expectations.

**Reply by joseacl** [reply]
Yes, absolutely, Thanks a lot.
